plugins/foreman_chef/0.1/index.md

```diff
@@ -0,0 +1,474 @@
+---
+layout: plugin
+title: Foreman Chef 0.1 Manual
+version: 0.1
+---
+
+# 1. {{ page.title }}
+
+This plugin enables Foreman to receive reports and attributes from chef clients
+and use these data for usual Foreman features like displaying status of your
+infrastructure on dashboard, monitoring based on convergence report, 
+boostraping chef-client during host provisioning etc. This plugin also make
+active communication with chef server possible so we can delete nodes and
+clients upon host deletion in Foreman.
+
+All communication is through Foreman Proxy which is running a [separate plugin](https://github.com/theforeman/smart_proxy_chef) and has
+direct access to the Chef server and Foreman instance. Chef clients must be
+able to connect to Foreman Proxy while the opposite direction is not required. The upload is triggered at the end of each chef client run, which is managed by [third plugin](https://github.com/theforeman/chef-handler-foreman) in form of chef handler, also provided by Foreman.
+
+![Infrastructure overview](infrastructure_schema.svg)
+
+Foreman Chef does support "legacy" mode where communication is direct
+with no Foreman Proxy involved. In this mode we do not handle authentication
+of any uploads from Chef clients, therefore it's strongly discouraged.
+
+## 1.1 Supported platforms
+
+TODO (1.9 on smart proxy, foreman-tasks support)
+
+# 2. Installation
+
+There are two different components to install: Foreman Chef plugin and
+Foreman Proxy Chef plugin (aka Smart Proxy Chef). Since Foreman 1.8 the
+installer supports both and it's failry easy to setup them. Supposing
+you added Foreman repositories and installed foreman-installer package you
+can start Foreman installation using this command
+
+    # to install Foreman with foreman_chef plugin
+    foreman-installer --enable-foreman-plugin-chef --enable-foreman-plugin-tasks
```

Comment:
Completely agree, ideas welcome, I hoped to have scenarios support in kafo already but it does not seem to be worked on anytime soon.